aesara/scalar/math.py

```diff
@@ -997,3 +998,59 @@ def c_code_cache_version(self):
 
 
 softplus = Softplus(upgrade_to_float, name="scalar_softplus")
+
+
+class Log1mexp(UnaryScalarOp):
+    r"""
+    Compute log(1 - exp(x)), also known as log1mexp
+
+    This function is numerically more stable than the naive approach.
+
+    For details, see
+    https://cran.r-project.org/web/packages/Rmpfr/vignettes/log1mexp-note.pdf
+
+    References
+    ----------
+    .. [Machler2012] Martin Mächler (2012).
+        "Accurately computing `\log(1-\exp(- \mid a \mid))` Assessed by the Rmpfr package"
+    """
+
+    @staticmethod
+    def static_impl(x):
+        if x < np.log(0.5):
+            return np.log1p(-np.exp(x))
+        else:
+            return np.log(-np.expm1(x))
+
+    def impl(self, x):
+        return Log1mexp.static_impl(x)
+
+    def grad(self, inp, grads):
+        (x,) = inp
+        (gz,) = grads
+        return [gz * true_div(1.0, 1.0 - exp(-x))]
+
+    def c_code(self, node, name, inp, out, sub):
+        (x,) = inp
+        (z,) = out
+
+        if node.inputs[0].type in float_types:
+            if node.inputs[0].type == float64:
+                return (
+                    """
+                    %(z)s = %(x)s < -0.6931471805599453 ? log1p(-exp(%(x)s)) : log(-expm1(%(x)s));
+                    """
+                    % locals()
+                )
+            else:
+                return (
+                    """
```

Comment:
This one should have been triggered by the tests in `tensor/test_math_scipy.py::TestLog1mexpBroadcast` when running on `floatX == "float32"` mode.

Reply:
The difference is just the `float` literal character?  We should simplify that to a condition in an f-string, especially since we're not even using that old string-interpolation syntax anymore (except in code that still hasn't been updated).

Reply:
> The difference is just the `float` literal character? We should simplify that to a condition in an f-string, especially since we're not even using that old string-interpolation syntax anymore (except in code that still hasn't been updated).

What syntax are we using now for inline c-code?

Reply:
> Edit: I guess an f-string instead of the `locals()`

Yeah, that's all I was talking about.